reco_utils/recommender/deeprec/io/nextitnet_iterator.py

```diff
@@ -0,0 +1,269 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import tensorflow as tf
+import numpy as np
+import json
+import pickle as pkl
+import random
+import os
+import time
+
+from reco_utils.recommender.deeprec.io.sequential_iterator import SequentialIterator
+from reco_utils.recommender.deeprec.deeprec_utils import load_dict
+
+__all__ = ["NextItNetIterator"]
+
+
+class NextItNetIterator(SequentialIterator):
+    """Data loader for the NextItNet model.
+    NextItNet requires a special type of data format. In training stage, each instance will produce (sequence_length * train_num_ngs) target items and labels, to let NextItNet output predictions of every item in a sequence except only of the last item.
+    """
+
+    def __init__(self, hparams, graph, col_spliter="\t"):
+        """Initialize an iterator. Create necessary placeholders for the model.
+        * different from sequential iterator
+        
+        Args:
+            hparams (obj): Global hyper-parameters. Some key settings such as #_feature and #_field are there.
+            graph (obj): the running graph. All created placeholder will be added to this graph.
+            col_spliter (str): column spliter in one line.
+        """
+        self.col_spliter = col_spliter
+        user_vocab, item_vocab, cate_vocab = (
```

Comment:
Doesn't this create an unnecessary tuple and unwrap back? 🤔 
Also this doesn't make the code shorter.

Reply:
fixed